### PR TITLE
Add doc examples for FileType struct

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #29356.

r? @steveklabnik

### CODE REVIEW

File: src/libstd/fs.rs [hunk 1]
```diff
@@ -788,14 +804,56 @@ impl Permissions {
...
+    /// # Examples
+    ///
+    /// ```
+    /// # fn foo() -> std::io::Result<()> {
+    /// use std::fs;
+    ///
+    /// let metadata = try!(fs::metadata("foo.txt"));
+    /// let file_type = metadata.file_type();
+    ///
+    /// assert!(file_type.is_file() == true);
```
Comment (frewsxcv) [inline src/libstd/fs.rs]: Alternatively, just `assert!(file_type.is_file());`

Comment (GuillaumeGomez) [inline src/libstd/fs.rs]: I like being explicit. :)

Comment (frewsxcv) [inline src/libstd/fs.rs]: If you want to keep the `==`, why not use `assert_eq!`?

Comment (GuillaumeGomez) [inline src/libstd/fs.rs]: Good idea!

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**